okapi-core/src/main/java/org/folio/okapi/util/ModuleUtil.java

```diff
@@ -167,4 +168,37 @@ public static String moduleList(List<ModuleDescriptor> moduleDescriptors) {
     }
     return str.toString();
   }
+
+  /**
+   * Return list of obsolete modules.
+   *
+   * <p>A module is considered obsolete if it's a snapshot *and* there is a higher
+   * non-snapshot version.
+   * @param mdl list of modules to consider
+   * @return list of obsolete modules.
+   */
+  public static List<ModuleDescriptor> getObsolete(List<ModuleDescriptor> mdl) {
+    mdl.sort(Collections.reverseOrder());
```

Comment:
This sorts the input list in descending order. This should be documented in the javadoc so that the caller knows to pass a modifiable list and can make a copy if the old order is still needed.